libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -979,3 +1018,56 @@ eth::AssemblyPointer ContractCompiler::cloneRuntime() const
 	a << u256(0x20) << u256(0) << Instruction::RETURN;
 	return make_shared<eth::Assembly>(a);
 }
+
+void ContractCompiler::addScopedVariable(VariableDeclaration const& _decl)
+{
+	// TODO check > 17
+	m_scopedVariables[_decl.scope()].push_back(&_decl);
+	if (!m_loops.empty())
+		m_loopScopedVariables[m_loops.top()].push_back(&_decl);
+	appendStackVariableInitialisation(_decl);
+}
+
+void ContractCompiler::popBlockScopedVariables(ASTNode const* _node)
+{
+	for (auto _decl : m_scopedVariables[_node])
+	{
+		CompilerContext::LocationSetter locationSetter(m_context, *_decl);
+		m_context << Instruction::POP;
```

Comment:
This should be `CompilerUtils utils(m_context).popStackElement(_decl->type())` - not all types occupy exactly one stack slot.